tests/foreman/ui/test_http_proxy.py

```diff
@@ -244,6 +246,59 @@ def test_set_default_http_proxy(module_org, module_location, setting_update, tar
         assert result['table'][0]['Value'] == "Empty"
 
 
+@pytest.mark.tier1
+@pytest.mark.run_in_one_thread
+@pytest.mark.parametrize('setting_update', ['content_default_http_proxy'], indirect=True)
+def test_positive_set_default_http_proxy(
+    request, module_org, module_location, setting_update, target_sat
+):
+    """Setting "Default HTTP proxy" when new HTTP proxy is created.
+
+    :id: e93733e1-5c05-4b7f-89e4-253b9ce55a5b
+
+    :steps:
+        1. Navigate to Infrastructure > Http Proxies
+        2. Create a Http Proxy and set "Default content HTTP proxy"
+        3. Navigate to Administer > Settings > Content tab
+        4. Verify the "Default HTTP Proxy" setting with created above.
+        5. Update "Default HTTP Proxy" to "no global default".
+
+    :Verifies: SAT-28860
+
+    :expectedresults: Creating Http Proxy with option "Default content HTTP proxy",
+        updates setting "Default HTTP Proxy" succesfully.
+    """
+    property_name = setting_update.name
+    http_proxy_name = gen_string('alpha', 15)
+    http_proxy_url = settings.http_proxy.un_auth_proxy_url
+
+    with target_sat.ui_session() as session:
+        session.http_proxy.create(
+            {
+                'http_proxy.name': http_proxy_name,
+                'http_proxy.url': http_proxy_url,
+                'http_proxy.content_default_http_proxy': 'true',
+                'locations.resources.assigned': [module_location.name],
+                'organizations.resources.assigned': [module_org.name],
+            }
+        )
+
+        # Teardown
+        @request.addfinalizer
+        def _finalize():
+            target_sat.api.HTTPProxy().search(query={'search': f'name={http_proxy_name}'})[
+                0
+            ].delete()
+            default_proxy = target_sat.api.Setting().search(
+                query={'search': 'name=content_default_http_proxy'}
+            )[0]
+            assert default_proxy.value != http_proxy_name
```

Comment:
It's not a good practice to do assertions in teardown. Can we handle assertions in the test and keep the cleanup part separate? The results are misleading when debugging failures.

Reply:
I don't think so it'll mislead us while debugging any failures, instead I think teardowns are best to ensure the Satellite we're using is in the same state as it was before, and adding assertions there, ensures everything worked in setup and teardown